### PR TITLE
Add `min` aggregation function

### LINKED ISSUE #2913: Support for `min` aggregation function while summarizing columns
## Problem
Currently, two aggregation functions (**list & count**) are supported as transformation steps while summarizing columns. Besides, there are many useful aggregation functions and `min` is also one of them.

## Proposed solution
It would be great to support `min` as a transformation step while summarizing columns.

## Additional context
This feature request is a part of GSoC '23 project, [Add More Summarization Functions](https://summerofcode.withgoogle.com/programs/2023/projects/D9XvkQCN).

### PR DESCRIPTION
**Pull request by Aritra8438**
Fixes #2913  

**This pull request adds support for `Min` aggregation as a transformation step in the data explorer.** 

User will be able to:
- Get `Min` of the groups based on the `Group By` clause (for any column type).
---
**Technical details**

I have made the following changes:
- Add `Min(DBFunction)` class to support min aggregation
- Add `min` suffix to the output alias of the aggregated column.
- Add `Min` to the list of transformation steps in the data explorer.

## Checklist
 
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the `develop` branch of the repository
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by pavish** [comment]
@Aritra8438 Please stack this PR on top of https://github.com/centerofci/mathesar/pull/2893, since @seancolsen's making some frontend changes there which would also apply to this one.

**Comment by rajatvijay** [comment]
@Aritra8438 it means

1. Create a new branch from `Aritra8438:add-support-for-sum-aggregation-function` and do your changes on top of it. 
2. When creating the PR, the base should be `Aritra8438:add-support-for-sum-aggregation-function`

**Comment by seancolsen** [comment]
@Aritra8438,

@rajatvijay said:

> When creating the PR, the base should be `Aritra8438:add-support-for-sum-aggregation-function`

It's more than that. The _commits_ need to be based off that branch too (not just the PR). Otherwise the PR will have all sorts of unrelated changes. @Aritra8438 if you created your git branch from `develop`, you'll need to rebase it. @mathemancer or I can help you with this over a call if needed. The exact steps will depend a little on what else is going on.

**Comment by Aritra8438** [comment]
> The _commits_ need to be based off that branch too (not just the PR). 

- For new branches,
```python
$ git checkout add-support-for-sum-aggregation-function

# some changes
$ git checkout -b add-support-for-new-aggregation-function

$ git add.
$ git commit -m "message"

$ git push origin add-support-for-new-aggregation-function
```
And then base of the PR will be `add-support-for-sum-aggregation-function`

- For existing braches (like this one):
```python
$ git checkout add-support-for-min-aggregation-function # (Based on upstream(/local) develop)

$ git rebase -i add-support-for-sum-aggregation-function

# Organize commits, solve conflicts
$ git rebase --continue

$ git push origin add-support-for-main-aggregation-function
```
And then base of the PR will be `add-support-for-sum-aggregation-function`

@seancolsen, is this the workflow? 
It would be helpful if I get it clarified from you and @mathemancer on Thursday.

**Comment by seancolsen** [comment]
@Aritra8438 You've got the top part correct.

The rebase stuff can get more complex though. In my experience mentoring other devs with rebase, I've seen things go awry several times before. There are different ways that it can go wrong, with different ways of fixing them. For all its wonders, Git really does not make this sort of thing easy. Rebase is a very powerful feature that I use at least several times per week. But it's important to understand how it works before using it, which is why I recommended a call. Feel free to reach out to me or Brent on Matrix for help if you like. Otherwise, waiting until Thursday is fine too.

**Review by mathemancer — APPROVED** [review]
Back end changes are fine. Please sort out the front end changes, etc.

**Comment by Aritra8438** [comment]
Hi @mathemancer and @seancolsen, **it's actually stacked on PR** #2912. 

But, I can't change the base as it's from my fork. 
So, merging #2912 and then updating this branch will make it ready for merge as well.

File: mathesar_ui/src/stores/abstract-types/operations/summarization.ts [hunk 1]
```diff
@@ -56,6 +56,14 @@ const functionsResponse: AbstractTypeSummarizationFunctionsResponse = {
...
     },
   },
+  max: {
+    label: 'Max',
+    inputOutputTypeMap: mapInputTypesToTheSameOutputType(),
+  },
+  min: {
+    label: 'Max',
+    inputOutputTypeMap: mapInputTypesToTheSameOutputType(),
+  },
```
Comment (seancolsen) [inline mathesar_ui/src/stores/abstract-types/operations/summarization.ts:66]: Please change the label of the `min` function from `'Max'` to `'Min'`.

**Review by seancolsen — CHANGES_REQUESTED** [review]

File: mathesar_ui/src/stores/abstract-types/operations/summarization.ts (hunk 1 above)
Comment (Aritra8438) [reply mathesar_ui/src/stores/abstract-types/operations/summarization.ts]: Sorry @seancolsen. Missed this while rebasing. PTAL!

**Comment by kgodey** [comment]
Just a quick note that you now have write permissions to the repo, so you don't need to use a fork, you can just checkout and push to a branch in this repo. 

That's probably not useful for this PR, but you might want to do that for future PRs.

**Comment by Aritra8438** [comment]
> you can just checkout and push to a branch in this repo.

Yes @kgodey, I have done the same for my recent 3 PRs (e.g. this one -> #2945)

**Review by seancolsen — APPROVED** [review]

**Pull request merged**